libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1682,6 +1682,7 @@ IotMqttError_t IotMqtt_Wait( IotMqttOperation_t operation,
     if( _IotMqtt_ValidateOperation( operation ) == false )
     {
         status = IOT_MQTT_BAD_PARAMETER;
+        goto cleanup;
```

Comment:
The commit message isn't accurate since we are changing functionality here.  Prior to this change `pMqttConnection` would have been updated on line 1689.  I'd either split this into two separate commits, or change the description to say "Fixed error handling case in IotMqtt_Wait and increased code coverage".